cmd/spire-server/cli/run/run.go

```diff
@@ -118,26 +125,26 @@ func parseFile(filePath string) (*RunConfig, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	if err := hcl.DecodeObject(&c, hclTree); err != nil {
 		return nil, err
 	}
 
 	return c, nil
 }
 
-func parseFlags(args []string) (*RunConfig, error) {
+func parseFlags(args []string) (*runConfig, error) {
 	flags := flag.NewFlagSet("run", flag.ContinueOnError)
-	c := &RunConfig{}
-
-	flags.StringVar(&c.BindAddress, "bindAddress", "", "IP address or DNS name of the SPIRE server")
-	flags.IntVar(&c.BindPort, "serverPort", 0, "Port number of the SPIRE server")
-	flags.IntVar(&c.BindHTTPPort, "bindHTTPPort", 0, "HTTP Port number of the SPIRE server")
-	flags.StringVar(&c.TrustDomain, "trustDomain", "", "The trust domain that this server belongs to")
-	flags.StringVar(&c.PluginDir, "pluginDir", "", "Plugin conf.d configuration directory")
-	flags.StringVar(&c.LogFile, "logFile", "", "File to write logs to")
-	flags.StringVar(&c.LogLevel, "logLevel", "", "DEBUG, INFO, WARN or ERROR")
-	flags.StringVar(&c.ConfigPath, "config", defaultConfigPath, "Path to a SPIRE config file")
-	flags.StringVar(&c.Umask, "umask", "", "Umask value to use for new files")
+	c := &runConfig{}
+
+	flags.StringVar(&c.Server.BindAddress, "bindAddress", defaultBindAddress, "IP address or DNS name of the SPIRE server")
```

Comment:
in merge config, we check to see if the cli config field is set to non-zero, and if it is then set it... we check first file, then cli config... so if we set the default here, and it has been configured in file and i do not specify again on cli here, the default will take over even if file is set for non-default. same for serverPort and bindHTTPPort below

Reply:
Got it. I missed that. Thanks!